### DISCUSSION TITLE
Discern between default value and provided value for None as a query parameter

### DISCUSSION THREAD

**Discussion by Subaku** [fastapi/fastapi#8941] [Questions]
Hello,

I've got a use case that sometimes comes up. When filtering my DB querying for "null" from the client on a particular field is a valid thing to do. This of course translates to `None`. Problem is most of the time query parameters of optional for me. How might I determine which `None` I'm dealing with? I.e. did the client explicitly provide the parameter or was it missing and `None` was set as the default. 

From similar issues I'm thinking that maybe some kind of middleware to determine if I actually passed a value or not might be in order but wanted to see if I'm missing something elegant.

Thanks!

**Comment by ilyasotkov** [comment]
@Subaku Did you find a solution/workaround for this?

My use case:

- `x_enable` is `False` by default
- `x_value` is `0` by default

If the user sets `x_value` to a value >0, `x_enable` can be assumed `True`, unless `x_enable` is *explicitly* set to `False`. So I need a way to know whether `x_enable` is `False` because it's the default or because it was supplied by the user.

**Comment by ilyasotkov** [comment]
I am able to solve this by getting the query param value from the raw request. Example:

```py
from fastapi import FastAPI, Request
from pydantic.validators import BOOL_FALSE

app = FastAPI()


@app.get("/")
def read_root(request: Request, x_enable: bool = False):
    if not x_enable:
        if request.query_params.get("x_enable", "").lower() in BOOL_FALSE:
            # We know that x_enable was set to false by the user
            ...
        else:
            # x_enable is False because it's the default
            ...
```
